src/main/kotlin/com/dobby/backend/application/usecase/ParticipantSignupUseCase.kt

```diff
@@ -0,0 +1,35 @@
+package com.dobby.backend.application.usecase
+
+import com.dobby.backend.application.mapper.SignupMapper
+import com.dobby.backend.infrastructure.database.repository.ParticipantRepository
+import com.dobby.backend.infrastructure.token.JwtTokenProvider
+import com.dobby.backend.presentation.api.dto.request.ParticipantSignupRequest
+import com.dobby.backend.presentation.api.dto.response.MemberResponse
+import com.dobby.backend.presentation.api.dto.response.signup.SignupResponse
+import com.dobby.backend.util.AuthenticationUtils
+import jakarta.transaction.Transactional
+import org.springframework.stereotype.Component
+
+@Component
+class ParticipantSignupUseCase (
+    private val participantRepository: ParticipantRepository,
+    private val jwtTokenProvider: JwtTokenProvider
+):UseCase<ParticipantSignupRequest, SignupResponse>
+{
+    @Transactional
+    override fun execute(input: ParticipantSignupRequest): SignupResponse {
+        val memberEntity = SignupMapper.toMember(input)
+        val participantEntity = SignupMapper.toParticipant(memberEntity, input)
+
+        val newParticipant = participantRepository.save(participantEntity)
+
+        val accessToken = jwtTokenProvider.generateAccessToken(AuthenticationUtils.createAuthentication(memberEntity))
+        val refreshToken = jwtTokenProvider.generateRefreshToken(AuthenticationUtils.createAuthentication(memberEntity))
```

Comment:
`createAuthentication(memberEntity)`를 두 번 호출해서 같은 인증 정보를 두 번 생성하는데 아래와 같이 인증 정보는 한 번만 생성하는 게 어떨까요??

```suggestion
        val authentication = AuthenticationUtils.createAuthentication(memberEntity)
        val accessToken = jwtTokenProvider.generateAccessToken(authentication)
        val refreshToken = jwtTokenProvider.generateRefreshToken(authentication)
```

Reply:
헉 커다란 실수를 할 뻔 했네요 🥲🥲
코드 꼼꼼하게 읽어주셔서 감사합니다! 수정 사항 반영하겠습니다!!